src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx

```diff
@@ -0,0 +1,52 @@
+"use client"
+
+import { useParams } from "next/navigation"
+import { PropsWithChildren, createContext, useContext, useEffect, useState } from "react"
+
+type CourseContextType = {
+    courseId: number | null
+    courseName: string
+}
+
+const CourseContext = createContext<CourseContextType>({
+  courseId: null,
+  courseName: '',
+})
+
+const useCourseProvider = (): CourseContextType => {
+  const { courseId } = useParams<{ courseId: string }>()
+  const [courseName, setCourseName] = useState<string>('')
+
+  useEffect(() => {
+    const fetchCourse = async () => {
+      try {
+        const courseResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/courses/${courseId}`,)
+        const courseData = await courseResponse.json()
+        setCourseName(courseData.name)
+      } catch (e) {
+        console.error(e)
+      }
+    }
+    if (courseId && !isNaN(Number(courseId))) {
+      fetchCourse()
+    }
+  }, [courseId, setCourseName])
+
+  return {
+    courseId: !isNaN(Number(courseId)) ? Number(courseId) : null,
```

Comment:
nit;

consider that you're doing this validation only partially. A valid course id would be a positive integer specifically, and this just validated that it's a number at all

Reply:
I'll address this in the next pr